otherlibs/fiber-lwt/fiber_lwt.ml

```diff
@@ -0,0 +1,32 @@
+open Stdune
+
+module Fiber_inside_lwt = struct
+  let key = Fiber.Var.create ()
+
+  let run fiber =
+    let fills, push = Lwt_stream.create () in
+    let fiber = Fiber.Var.set key push (fun () -> fiber) in
+    let rec loop = function
+      | Fiber.Scheduler.Done x -> Lwt.return x
+      | Fiber.Scheduler.Stalled stalled ->
+        Lwt.bind (Lwt_stream.next fills) (fun fill ->
+            loop (Fiber.Scheduler.advance stalled [ fill ]))
+    in
+    loop (Fiber.Scheduler.start fiber)
+
+  let callback_to_lwt f =
+    Fiber.bind (Fiber.Var.get key) ~f:(function
+      | None ->
+        failwith "Fiber_lwt.Fiber_inside_lwt.run_lwt: called outside [run]"
+      | Some push_fill ->
+        let ivar = Fiber.Ivar.create () in
+        Lwt.async (fun () ->
+            Lwt.bind
+              (Lwt.try_bind f
+                 (fun x -> Lwt.return (Ok x))
+                 (fun exn -> Lwt.return (Error exn)))
+              (fun x ->
+                push_fill (Some (Fiber.Fill (ivar, x)));
```

Comment:
When `Ivar.fill` returns unit, is it going to be possible to just call it directly here?

Reply:
I'm not planning to do that anymore. Making `Ivar.fill` return `unit Fiber.t` allows for a faster implementation of the scheduler in #5230

Reply:
I was thinking about this again yesterday. Even if `Ivar.fill` was returning `unit`, we would still need to notify the lwt loop in `run` so that it wakes up and try to advance the fiber. So it wouldn't simplify this code.